### PR TITLE
Rename probeloader to kernelmoduleloader

### LINKED ISSUE #53: Probe loader should be renamed
**What happened**:

Doing some homework on building an `eBPF` probe loader into `falcoctl` and I checked out the `probeloader` package.

It's currently using syscall `313` which is for loading kernel modules (not eBPF probes)

**What you expected to happen**:

Maybe we should call the package `kernelmoduleloader`?

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by AnishShah**
**What type of PR is this?**

/kind cleanup

**Any specific area of the project related to this PR?**

/area library

**What this PR does / why we need it**:

Cleanup
**Which issue(s) this PR fixes**:

Fixes #53 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
chore(cmd): rename falcoctl install probe to falcoctl install module
```

### CODE REVIEW

File: cmd/install_probe_linux.go [hunk 1]
```diff
@@ -108,15 +108,15 @@ func NewProbeInstallCommand(streams genericclioptions.IOStreams) *cobra.Command
 			}
 
 			// fetch module
-			err = probeloader.FetchModule(o.falcoProbeURL, falcoProbeFullpath)
+			err = kernelmoduleloader.FetchModule(o.falcoProbeURL, falcoProbeFullpath)
```
Comment (kris-nova) [inline cmd/install_probe_linux.go]: Looks like we are still using the word `probe` in this package. 

We need to be very clear about when we are referring to a kernel module, or a BPF probe - they are very different things

Comment (AnishShah) [reply cmd/install_probe_linux.go]: Sorry Kris. I didn't understand. You mean, refactoring loader.go file such that we separate functions related to BPF probe and kernel module into respective pkgs?

Comment (kris-nova) [reply cmd/install_probe_linux.go]: So there are two things, that require completely different logic.

1) Loading a kernel module such that Falco can use it to consume kernel events
2) Loading a BPF probe such that Falco can use it to consume kernel events

They are very different things, and (e)BPF is significantly more stable and reliable

---

Renaming the package is a good step in the broader effort here. 

We should only use the word `probe` if we are referring to anything related to (e)BPF

We should only use the word `module` if we are referring to anything related to a kernel module 

The `install_probe_linux.go` file is a misnomer. It does **not** install a BPF probe, it installs a kernel module (kind of). 

Having a file named `install_probe_linux.go` that installs a kernel module makes about as much sense as having a file named `install_emacs_linux.go` that installs actually installs `vim`. 

---

> Sorry Kris. I didn't understand. You mean, refactoring loader.go file such that we separate functions related to BPF probe and kernel module into respective pkgs?

Yes. Please have two packages. One for loading a BPF probe, and one for loading a kernel module. They are different things and require different logic, and should be named differently.

Comment (kris-nova) [reply cmd/install_probe_linux.go]: A rule of thumb here is that if we are dealing with a kernel module and the word `probe` appears -- there is a significant chance something is named incorrectly.

Comment (kris-nova) [reply cmd/install_probe_linux.go]: In order to close #53 the word `probe` should be completely removed from the codebase. 

Right now `falcoctl` doesn't do any `probe` loading. It however does `kernel module` loading.

If you want to refactor the word out, please do and we can merge this in :smile: 

Sorry to be such a pain - but there is a **LOT** of misnomers in Falco, and we have to start being very diligent about how we name things.

Comment (AnishShah) [reply cmd/install_probe_linux.go]: Thanks for the explanation, Kris. So we should also rename `install probe` command to `install module`.

File: cmd/install_module_linux.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+				return err
+			}
+
+			logger.Always("FALCO_VERSION: %s", o.falcoVersion)
+			logger.Always("FALCO_MODULE_URL: %s", o.falcoModuleURL)
+			logger.Always("FALCO_MODULE_REPO: %s", o.falcoModuleRepo)
+			logger.Always("KERNEL_VERSION: %s", falcoKernelRelease)
+			logger.Always("KERNEL_CONFIG_HASH: %s", falcoConfigHash)
+
+			// if FALCO_PROBE_URL not set, build it
```
Comment (kris-nova) [inline cmd/install_module_linux.go]: Can we adjust or delete this comment please

Comment (kris-nova) [inline cmd/install_module_linux.go]:
```suggestion
			// if FALCO_MODULE_URL not set, build it
```

File: pkg/kernelmoduleloader/loader.go [hunk 3]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
```
Comment (kris-nova) [inline pkg/kernelmoduleloader/loader.go]: Can we also rename this file from `loader.go` to `kernel_module_loader.go`?

We will more than likely have one for `bpf_probe_loader.go` in the future

**Comment by kris-nova** [comment]
Looks like we need a rebase :disappointed:

**Comment by AnishShah** [comment]
Hi Kris. I have rebased my branch.

**Comment by AnishShah** [comment]
Hi Kris. Any idea how to merge this? It says that it is in a merge pool

**Pull request merged**